event_model/documents/stream_datum.py

```diff
@@ -18,40 +32,37 @@ class StreamDatum(TypedDict):
             "be contiguous for a given stream.",
```

Comment:
@danielballan @tacaswell now we have `indices` I'm inclined to drop `block_idx` from `StreamDatum`, is that reasonable?

Reply:
Yes it is, as we are going to want to squash adjacent StreamDatums